tests/fixture/tmpnet/process_runtime.go

```diff
@@ -144,6 +154,51 @@ func (p *ProcessRuntime) Start(ctx context.Context) error {
 	return p.writeMonitoringConfig()
 }
 
+func (p *ProcessRuntime) writeFlags() error {
+	flags, err := p.node.composeFlags()
+	if err != nil {
+		return fmt.Errorf("failed to compose node flags: %w", err)
+	}
+
+	flags.SetDefaults(FlagsMap{
+		config.DataDirKey: p.node.DataDir,
+		// Use dynamic port allocation
+		config.HTTPPortKey:    "0",
+		config.StakingPortKey: "0",
+		// Binding to localhost on macos avoids having a permission
+		// dialog pop up for every node that tries to bind to
+		// non-localhost interfaces
+		config.PublicIPKey:    "127.0.0.1",
+		config.HTTPHostKey:    "127.0.0.1",
+		config.StakingHostKey: "127.0.0.1",
+	})
+
+	runtimeConfig := p.getRuntimeConfig()
+
+	// Only configure the plugin dir with a non-empty value to ensure the use of the
+	// default value (`[datadir]/plugins`) when no plugin dir is configured.
+	if len(runtimeConfig.PluginDir) > 0 {
```

Comment:
nit prefer using string checks to highlight it's a string without having to rely on the IDE showing what type it is

```suggestion
	if runtimeConfig.PluginDir != "" {
```